src/esp/io/JsonEspTypes.cpp

```diff
@@ -146,16 +167,43 @@ bool fromJsonValue(const JsonGenericValue& obj,
     const std::string shaderTypeLC =
         Cr::Utility::String::lowercase(shaderTypeToUseString);
     auto mapIter = metadata::attributes::ShaderTypeNamesMap.find(shaderTypeLC);
-    ESP_CHECK(
-        mapIter != metadata::attributes::ShaderTypeNamesMap.end(),
-        "Illegal shader_type value"
-            << shaderTypeToUseString
-            << "specified in JSON to be used to set AssetInfo.shaderTypeToUse. "
-               "Aborting.");
+    ESP_CHECK(mapIter != metadata::attributes::ShaderTypeNamesMap.end(),
+              "Illegal shader_type value '"
+                  << shaderTypeToUseString
+                  << "' specified in JSON to be used to set "
+                     "AssetInfo.shaderTypeToUse. "
+                     "Aborting.");
     x = mapIter->second;
   }
   return shaderTypeSucceess;
 }
 
+JsonGenericValue toJsonValue(const esp::gfx::LightPositionModel& x,
+                             JsonAllocator& allocator) {
+  return toJsonValue(metadata::attributes::getLightPositionModelName(x),
+                     allocator);
+}
+
+bool fromJsonValue(const JsonGenericValue& obj,
```

Comment:
Ug, serializing enums as strings like this is *so* much boilerplate code. We should find a better long-term solution.

Reply:
Unhelpful suggestion: store it as numbers? glTF does the same for various data types, sampler modes etc, and that simplifies parsing and writing tremendously. A downside is that you have to ensure that the numbering doesn't change in the future, and the JSON is not that much human-readable after.

Reply:
I think that this would make sense. It would yield smaller json files and would simplify the code. We can write unit tests to catch issues with backwards compatibility.

Reply:
Hi!  FWIW all the values backed by string->enum maps in the metadata were so chosen to facilitate user field modification in the JSON itself, which was expected to occur in the various config files these enum-backed string fields were supported in.  

I'm not sure if that is still the intent for these fields, but understand that changing the json values of any of these from strings to ints, while addressing some cumbersome parsing code, may also invalidate many extant config files, both in datasets and in use by folks.

Reply:
In this case, it's probably better to keep things as-is for now.

In the future, we could include a version to the json files so that we can run them through a backwards compatibility layer.

Reply:
Until then, it would also be very useful, i think, to put these the String->enum mapping function right next to the enum->string versions in metadata::attributes namespace (AttributesEnumMaps.h/cpp file).